### ISSUE TITLE
Due for macOS notifies of reminders that have already been cleared or postponed on another device.

### ISSUE THREAD

**Issue by uturnr** [dueapp/Due-macOS#100]
I often receive Due reminders on my phone and then delay them or mark them complete. Then moments later, I start getting the same reminder on my computer, repeating every minute. I then manually perform a sync to update the macOS client. (I sync with iCloud)

I'm not sure if there are limitations to how often iCloud can be used for syncing, but to have the app on my Mac is only useful for me if it is more in sync with my phone. Ideally, I would want the macOS client to always run a sync before triggering a notification, including repeat notifications.

**Comment by junjie** [comment]
There are 2 possible reasons why background sync can fail on the Mac. One, the silent push used to communicate that a change has happened was not sent out. Two, the silent push was sent out, but was not picked up by the app.

To ensure that the pushes are being sent out:

1. Ensure Sync in Background is enabled on your iOS device > Due > Settings > Sync.
2. Leave Due open on your Mac and iPhone (unlocked and screen active). Make a change on your Mac, manually trigger a sync and wait for upload to complete. Does the iPhone pick up the change automatically?

If (2) happens properly, i.e. iPhone is picking up the changes, then we've determined that the pushes were sent out.

That leaves us with the other possibility, that the silent push was not picked up by the app. 

On macOS, you may sometimes encounter a bug in macOS where the push was successfully delivered to the system, but yet the system did not pass the notification over to Due.

There are some ways to resolve the problem, in increasing order of inconvenience. So perform each step, then try to make a change on your iPhone, trigger a sync in Settings > Sync, wait for upload to complete, and see if the Mac picks up the change. If not, proceed to the next step:

1. Relaunch the Mac app, try again.
2. Unlink from the sync service on Mac app, relink with the sync service, try again.
3. Restart your computer, try again.
4. Delete the Due Mac app, reinstall from App Store, relink with sync service, try again.

If it did not sync (almost) immediately after the changes have been uploaded on your iPhone/iPad, that means it will never pick up the change by itself. So you don't have to wait too long to know that it is not working, and to move on to the next step.

**Comment by uturnr** [comment]
Thanks @junjie - The notifications appear to be syncing smoothly after step 4 - delete, reinstall, relink.
